scenarios.md

```diff
@@ -0,0 +1,199 @@
+
+1. A CI solution responds to a git commit notification
+1. The CI system clones the git repo and builds the artifacts, with fully qualified names: **image**: `myregistry.domain.com/sample/hello-world:aa31` and a **helm chart**: `myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The CI system signs the artifact with locally available keys *
+  (**Note:** key management deferred from this requirement)*
+1. The artifact is pushed to a registry:  
+  `$ docker push myregistry.domain.com/sample/hello-world:aa31`
+  `$ helm chart push myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The artifacts are deployed, referencing the registry they were pushed to.
+
+**Implications of this requirement:**
+
+- nothing additional from previous scenarios.
+
+### Scenario #4: Promote Artifacts Within a Registry, Using a Different repo
+
+A CI/CD system promotes validated artifacts from a dev repository to production repositories.
+
+1. A CI/CD solution responds to a git commit notification, cloning, building, signing and pushing the artifacts to a development repo within their registry.
+1. As the CI/CD solution runs functional, determining the artifacts are ready for production, the artifacts are moved from one repo to another.  
+  `$ docker tag myregistry.domain.com/dev/alpha-team/web:1abc myregistry.domain.com/prod/web:1abc`
+
+**Implications of this requirement:**
+
+- Renaming does not invalidate the signature
+- Artifact movement, or copy, to a different repository does not invalidate the signature
+
+### Scenario #5: Acquire Artifacts From Public Repositories, Copying to Private Registries for Deployment
+
+As artifacts evolve from base images developers build upon, to products that customers run directly, they will need to run them from private registries which maintain and validate the signature provided by the software vendor.
+
+1. ACME Rockets acquires networking monitoring software they run in their environment.
+1. ACME Rockets copies the artifacts, renaming them to reflect their private registry, and pushes them to their registry.  
+  `$ docker pull registry.wabbit-networks.com/netmonitor:1.0`
+  `$ docker tag registry.wabbit-networks.com/netmonitor:1.0 registry.acme-rockets.com/infra/networking/wabbit-net-monitor:1.0`
+  `$ docker push registry.acme-rockets.com/infra/networking/wabbit-net-monitor:1.0`
+1. As ACME Rockets deploys the networking monitoring software, the signatures are validated to be signed by ACME Rockets.
+
+**Implications of this requirement:**
+
+- In addition to the above scenario where an artifact is promoted within the same registry, this scenario requires the signature to be available in the new registry. Whether the signature is moved, or referenced from it's original location is a subject for design discussion. In this scenario, ACME Rockets could validate the signature against a wabbit-network signature server.
```

Comment:
> from it's original

from _its_ original

Reply:
grr, I'm trying to hard to catch these...